paymentapi/src/test/java/fi/hel/verkkokauppa/payment/api/PaymentAdminControllerUnitTest.java

```diff
@@ -91,4 +145,310 @@ public void savePaymentFilter() throws Exception {
         Assertions.assertEquals(expected.getFilterType(), paymentFilterDto.getFilterType());
         Assertions.assertEquals(expected.getValue(), paymentFilterDto.getValue());
     }
-}
\ No newline at end of file
+
+
+    /**
+     * It tests the create payment method endpoint.
+     */
+    @Test
+    public void whenCreatePaymentMethodWithValidData_thenReturnStatus201() throws Exception {
+        PaymentMethodDto paymentMethodDto = createTestPaymentMethodDto(GatewayEnum.OFFLINE);
+        PaymentMethod paymentMethod = mapper.convertValue(paymentMethodDto, PaymentMethod.class);
+
+        Mockito.when(paymentMethodService.createNewPaymentMethod(paymentMethodDto)).thenCallRealMethod();
+        Mockito.when(paymentMethodRepository.save(paymentMethod)).thenReturn(paymentMethod);
+
+        MvcResult response = this.mockMvc.perform(
+                        post("/payment-admin/payment-method")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .accept(MediaType.APPLICATION_JSON)
+                                .content(getFileContents("data/paymentMethodCreateRequest.json"))
+                )
+                .andDo(print())
+                .andExpect(status().is2xxSuccessful())
+                .andExpect(status().is(201))
+                .andReturn();
+
+        // TODO: Fix below request body verification - body is empty for even if status is 201 (?)
```

Comment:
This should be removed or be fixed?

Reply:
I added this here as a comment in case some of you would see what I have done wrong for the test to fail. But I'll check it out one more time and remove it if I don't get it to work.